### PR TITLE
Remove `service_has permission` function

### PR DESCRIPTION
**Pull request by quis**
Using the method on the service model itself is less verbose.

Some detail of specific changes in individual commits.

### CODE REVIEW

File: app/template/rest.py [hunk 1]
```diff
@@ -80,12 +80,12 @@ def validate_parent_folder(template_json):
 def create_template(service_id):
     fetched_service = dao_fetch_service_by_id(service_id=service_id)
     # permissions needs to be placed here otherwise marshmallow will interfere with versioning
-    permissions = [p.permission for p in fetched_service.permissions]
+    [p.permission for p in fetched_service.permissions]
```
Comment (BlessedDev) [inline app/template/rest.py:83]: This line seems to be an  accidental holdover

**Review by BlessedDev — DISMISSED** [review]
The changes make sense and makes the code not just less verbose but more readable.
I have flagged up one issue for your consideration.

File: app/template/rest.py (hunk 1 above)
Comment (quis) [reply app/template/rest.py:83]: Weirdly when I remove that line the tests fail. Probably related to the comment on the line above. Will see if I can investigate and work out why this is happening

Comment (quis) [reply app/template/rest.py:83]: I have done some digging into why this was happening.

Here’s a fix, which should be merged first: https://github.com/alphagov/notifications-api/pull/4048

Then in this PR we can remove the line which is seemingly doing nothing.

**Review by leohemsted — APPROVED** [review]

**Pull request merged**